src/d3a/d3a_core/sim_results/endpoint_buffer.py

```diff
@@ -61,6 +78,21 @@ def __init__(self, job_id, initial_params, area, should_export_plots):
                 ConstSettings.GeneralSettings.EXPORT_ENERGY_TRADE_PROFILE_HR:
             self.offer_bid_trade_hr = OfferBidTradeGraphStats()
 
+        if constants.KAFKA_MOCK:
+            self.producer = Mock()
```

Comment:
Can you please clarify this? Production code is generally agnostic of testing functionalities (that's one reason why we wanted to remove all the `ast.literal_eval()` calls for scenarios).

Reply:
One possible way to achieve this would be to add `producer` to the `__init__` arguments. Then in tests you could do something like
```python
epb = SimulationEndpointBuffer("1", {"seed": 0}, grid, True, producer=Mock())
```